src/VisualStudio/VisualStudioDiagnosticsToolWindow/Loggers/OutputWindowLogger.cs

```diff
@@ -41,97 +43,28 @@ public bool IsEnabled(FunctionId functionId)
-            }
-
-            public OutputPane()
-            {
-                _serviceProvider = ServiceProvider.GlobalProvider;
-
-                var componentModel = (IComponentModel)_serviceProvider.GetService(typeof(SComponentModel));
-                _threadingContext = componentModel.GetService<IThreadingContext>();
-            }
-
-            private IVsOutputWindowPane _doNotAccessDirectlyOutputPane;
-
-            private void WriteLineInternal(string value)
-            {
-                var pane = GetPane();
-                if (pane == null)
-                {
-                    return;
-                }
-
-                pane.OutputStringThreadSafe(value + Environment.NewLine);
-            }
-
-            private IVsOutputWindowPane GetPane()
-            {
-                if (_doNotAccessDirectlyOutputPane == null)
-                {
-                    _threadingContext.JoinableTaskFactory.Run(async () =>
-                   {
-                       await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                       if (_doNotAccessDirectlyOutputPane != null)
-                       {
-                           // check whether other one already initialized output window.
-                           // the output API already handle double initialization, so this is just quick bail
-                           // rather than any functional issue
-                           return;
-                       }
-
-                       var outputWindow = (IVsOutputWindow)_serviceProvider.GetService(typeof(SVsOutputWindow));
-
-                       // this should bring outout window to the front
-                       _doNotAccessDirectlyOutputPane = CreateOutputPane(outputWindow);
-                   });
-                }
-
-                return _doNotAccessDirectlyOutputPane;
-            }
-
-            private IVsOutputWindowPane CreateOutputPane(IVsOutputWindow outputWindow)
+            var asyncToken = _asyncListener.BeginAsyncOperation(nameof(WriteLine));
+            Task.Run(async () =>
             {
-                _threadingContext.ThrowIfNotOnUIThread();
-
-                // Try to get the workspace pane if it has already been registered
-                var workspacePaneGuid = s_outputPaneGuid;
-
-                // If the pane has already been created, CreatePane returns it
-                if (ErrorHandler.Succeeded(outputWindow.CreatePane(ref workspacePaneGuid, "Roslyn Logger Output", fInitVisible: 1, fClearWithSolution: 1)) &&
-                    ErrorHandler.Succeeded(outputWindow.GetPane(ref workspacePaneGuid, out var pane)))
-                {
-                    return pane;
-                }
-
-                return null;
-            }
+                using var outputChannelStore = await _serviceBrokerClient.GetProxyAsync<IOutputChannelStore>(VisualStudioServices.VS2019_4.OutputChannelStore).ConfigureAwait(false);
```

Comment:
Same here: I'm unfamiliar with this using syntax but it appears you're disposing the rental before using it. That would be a bug.